### ISSUE TITLE
[emesene-2.12.5] retrieving chat history leads crash

### ISSUE THREAD

**Issue by pengshao** [emesene/emesene#1437]
Hello, thanks for reading my bug report. I am using emesene-2.12.5, under a gentoo amd64 machine. I noticed that when I selected a contact in MSN, and choose view information-->chat history, after a few seconds the emesene crashed.  I tried to refresh my profile following the instruction on your blog but it didn't work.

I tried to read the console output, but it appears that following error message appeared the most often

```
/usr/lib64/python2.7/site-packages/emesene/gui/gtkui/__init__.py:102: GtkWarning: Invalid text buffer iterator: either the iterator is uninitialized, or the characters/pixbufs/widgets in the buffer have been modified since the iterator was created.
You must use marks, character numbers, or line numbers to preserve a position across buffer modifications.
You can apply tags and insert marks without invalidating your iterators,
but any mutation that affects 'indexable' buffer contents (contents that can be referred to by character offset)
will invalidate all outstanding iterators
  gtk.main()
/usr/lib64/python2.7/site-packages/emesene/gui/gtkui/RichBuffer.py:108: GtkWarning: gtk_text_layout_real_invalidate: assertion `layout->wrap_loop_count == 0' failed
  self.insert_with_tags(iterator, text, *tags)
/usr/lib64/python2.7/site-packages/emesene/gui/gtkui/RichBuffer.py:110: GtkWarning: gtk_text_layout_real_invalidate: assertion `layout->wrap_loop_count == 0' failed
  self.insert(iterator, text)
emesene-2: Fatal IO error 11 (Resource temporarily unavailable) on X server :0.0.
Segmentation fault
```

However, the following error message was also observed a couple of times

```
[xcb] Unknown request in queue while dequeuing
[xcb] Most likely this is a multi-threaded client and XInitThreads has not been called
[xcb] Aborting, sorry about that.
python2: /var/tmp/portage/x11-libs/libX11-1.5.0/work/libX11-1.5.0/src/xcb_io.c:178: dequeue_pending_request: Assertion `!xcb_xlib_unknown_req_in_deq' failed.
Aborted
```

I tested 2.12.6-dev, the problem remains. 

Thanks for taking care of it.

**Comment by Sbte** [comment]
Are you using "Output Text" under "conversation output" in Preferences->Extensions? In this case I have now clue what's going on.

If you're using "Adium Output" instead, it's probably a duplicate of #1368

**Comment by pengshao** [comment]
Dear Sbte, rather odd, I have only the following options under Preference-->Extensions:

```
Conversation info ,  history exporter,  nice bar,  nice renderer, notifycationGUI,  notificationimage, tray icon
```

Under history exporter, there are two options: "save_logs_as_txt" and "save_logs_as_xml". Is this relevant?

Thanks.

**Comment by Sbte** [comment]
That just means that you don't have pywebkitgtk installed. It is highly recommended that you install this and use Adium Output.

Anyway, Output Text works fine here. Since you're using Gentoo, most likely the problem you're experiencing is a broken system.

**Comment by Sbte** [comment]
Oh, btw, since you're using Gentoo, try to rebuild glibc. That seems to work for a lot of bugs reported by Gentoo users.

**Comment by pengshao** [comment]
Dear Sbte, I just installed pywebkitgtk and the "conversation output" option appeared. Switching to "Adium Output" showed chat history successfully. I also tried to rebuild glibc but it didn't make "output text" successful.

I will notify the package maintainer of gentoo the pywebkitgtk dependency. Thanks for your instruction.

By the way, is there anyway to adjust the font size of Adium output? They are a little bit small on my computer and hard to read. I changed a couple of Adium themes but none of them allowed me to change the font size.  I am sorry if I shouldn't ask a how-to question in bug report but I didn't find anywhere to ask...
